### ISSUE TITLE
Required properties inside computed property name not detected

### ISSUE THREAD

**Issue by silverwind** [microsoft/TypeScript#61731]
### 🔎 Search Terms

"computed property name", "required" 

### 🕗 Version & Regression Information

- This changed between versions 5.1.6 and 5.2.0
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about (nonexistant)

### ⏯ Playground Link

https://www.typescriptlang.org/play/?ts=5.9.0-dev.20250519#code/MYewdgzgLgBAZiEAuGAlApqATgEwDzRYCWYA5gDQwDeYAhgLboqEmkC+AfDALzUBQMGAG0ABghAiAuiiptyA4SIAkVAEQAjWllVspMuQtEqAylGJkAFBtoAvVQEpd06gbZA

### 💻 Code

```ts
const foo: Record<string, {name: string}> = {
  [`foo`]: {},
  [`${"bar"}`]: {},
  [`${String("baz")}`]: {},
}
```

### 🙁 Actual behavior

The first and second properties are erroring as expected because of missing `name` property, but the third one does not, while it should.

### 🙂 Expected behavior

All three object properties should error because of missing `name` property.

### Additional information about the issue

In 5.1, only the first property error was detected, since 5.2, it now detects the first two. Ideally it should detect all three cases.

**Comment by RyanCavanaugh** [comment]
To limit error spam, TS intentionally doesn't try to compute more errors in cases where it appears that an error would simply be downstream of a different problem. This is one of those cases. You'll still see an error once the other cases are corrected:

```ts
// Error
const foo: Record<string, {name: string}> = {
  [`foo`]: { name: "" },
  [`${"bar"}`]: { name: "" },
  [`${String("baz")}`]: {},
}
```

```ts
// Also an error
const foo: Record<string, {name: string}> = {
  [`${String("baz")}`]: {},
}
```

**Comment by silverwind** [comment]
Thanks, you're right. I tried these variations and could not find a case where it actually missed reporting an error. The places where it reports them were a bit inconsistent, but it at least always reported it somewhere.
